src/wp-includes/class-wp-block.php

```diff
@@ -206,14 +206,9 @@ public function render( $options = array() ) {
 		if ( ! $options['dynamic'] || empty( $this->block_type->skip_inner_blocks ) ) {
 			$index = 0;
 			foreach ( $this->inner_content as $chunk ) {
-				if ( is_string( $chunk ) ) {
-					$block_content .= $chunk;
-				} else {
-					$parent_parsed_block  = $current_parsed_block;
-					$current_parsed_block = $this->inner_blocks[ $index ]->parsed_block;
-					$block_content       .= $this->inner_blocks[ $index++ ]->render();
-					$current_parsed_block = $parent_parsed_block;
-				}
+				$block_content .= is_string( $chunk ) ?
+					$chunk :
+					$this->inner_blocks[ $index++ ]->render();
 			}
 		}
 
```

Comment:
2 lines down here seems like the only place on Core where we call `render_callback` right? why not just add `WP_Block_Supports::$block_to_render = $this` there? instead of this filter https://github.com/WordPress/wordpress-develop/pull/640/files#r510866677